imagenameparser/api/image_name.py

```diff
@@ -0,0 +1,849 @@
+# Copyright (c) 2022 SUSE LLC
+#
+
+"""Provides the ``ImageName`` API class"""
+
+import logging
+
+from imagenameparser.api.image_name_regexp import ImageNameRegexp
+from imagenameparser.errors import BadRegexMatchError
+
+_logger = logging.getLogger(__name__)
+
+
+class ImageName:  # pylint: disable=R0904
+    """
+    Parse an image name.
+
+    ``framework`` should be a valid cloud provider framework name, such
+    as ``ec2`` or ``azure``.
+
+    ``name`` should be a valid image name, such as any of the image names
+    reported by ``pint``.
+
+    """
+
+    def __init__(self, framework, name):
+        self._framework = framework
```

Comment:
define a post_init() method and verify the framework argument is in the valid set, if not throw an exception.